### PR TITLE
Support Java 7 #81

### PR DESCRIPTION
**Pull request by markiewb**
Hi, have a look at this pull-request to transform the code to Java7. You did not use so much JDK8-specifics, so it was "easy" to transform. Try it out.

Using this you could provide a version for NB8.1/8.0 the plugin portal. https://github.com/welovecoding/editorconfig-netbeans/issues/42

### CODE REVIEW

File: nb-configuration.xml [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
+        <org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.text-limit-width>80</org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.text-limit-width>
+        <org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.text-line-wrap>none</org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.text-line-wrap>
+        <org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.wrapAnnotations>WRAP_IF_LONG</org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.wrapAnnotations>
+        <org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.sortMembersInGroups>true</org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.sortMembersInGroups>
+        <org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.keepGettersAndSettersTogether>true</org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.keepGettersAndSettersTogether>
+        <de-markiewb-netbeans-plugins-eclipse-formatter.preserveBreakPoints>true</de-markiewb-netbeans-plugins-eclipse-formatter.preserveBreakPoints>
+        <de-markiewb-netbeans-plugins-eclipse-formatter.eclipseFormatterActiveProfile/>
```
Comment (bennyn) [inline nb-configuration.xml]: I don't think that these entries `de-markiewb-netbeans-plugins-eclipse-formatter` are needed, are they?

**Comment by bennyn** [comment]
This is awesome news! I will review your pull request this weekend and then merge it in. Looks like we are going for our initial `1.0` release. Looking forward to it! Uh yeah! :sunglasses:

File: nb-configuration.xml (hunk 1 above)
Comment (markiewb) [inline nb-configuration.xml]: Die Zeilen waren bereits vorher drin. Nur irgendwie hat mein NB die Lineendings geaendert. Mein Change kann also raus!

**Comment by markiewb** [comment]
@bennyn: Schau dir besonders die Changes in FileInfoReader an! Den Code habe ich aus den Lambdas mit Readern nachgebaut. Hoffentlich funktioniert noch alles. Die restlichen Inline-Funktionen konnte mir NetBeans via Hint zurueckkonvertieren.

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java [hunk 2]
```diff
@@ -163,9 +164,53 @@ private static String readFirstLineWithSeparator(FileObject fo, Charset charset)
...
+
+  public static String trimTrailingWhitespace(String text, String lineEnding) {
+    List<String> lines = new ArrayList<>();
+    {
+      BufferedReader reader = new BufferedReader(new StringReader(text));
+      try {
+        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+          lines.add(line);
+        }
+        reader.close();
```
Comment (Shredder121) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: Shouldn't this `reader` be closed in a `finally` block or placed in a `try-with-resources` block?

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java [hunk 3]
```diff
@@ -163,9 +164,53 @@ private static String readFirstLineWithSeparator(FileObject fo, Charset charset)
...
+
+  public static String replaceLineEndings(String text, String lineEnding) {
+    List<String> lines = new ArrayList<>();
+    {
+      BufferedReader reader = new BufferedReader(new StringReader(text));
+      try {
+        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+          lines.add(line);
+        }
+        reader.close();
```
Comment (Shredder121) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: Shouldn't this `reader` be closed in a `finally` block or placed in a `try-with-resources` block?

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java (hunk 2 above)
Comment (junichi11) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: :+1:

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java (hunk 3 above)
Comment (junichi11) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: :+1:

**Comment by markiewb** [comment]
@Shredder121: I added try-with-resources. See the additional commit https://github.com/welovecoding/editorconfig-netbeans/commit/f738fee7e09c78e0ee3d478240d13aec4998bcb0 on the branch

File: nb-configuration.xml [hunk 4]
```diff
@@ -1,28 +1,29 @@
...
-        <de-markiewb-netbeans-plugins-eclipse-formatter.preserveBreakPoints>true</de-markiewb-netbeans-plugins-eclipse-formatter.preserveBreakPoints>
-        <de-markiewb-netbeans-plugins-eclipse-formatter.eclipseFormatterActiveProfile/>
-        <de-markiewb-netbeans-plugins-eclipse-formatter.eclipseFormatterLocation/>
-        <de-markiewb-netbeans-plugins-eclipse-formatter.showNotifications>false</de-markiewb-netbeans-plugins-eclipse-formatter.showNotifications>
-        <de-markiewb-netbeans-plugins-eclipse-formatter.eclipseFormatterEnabled>false</de-markiewb-netbeans-plugins-eclipse-formatter.eclipseFormatterEnabled>
-        <de-markiewb-netbeans-plugins-eclipse-formatter.enableFormatAsSaveAction>false</de-markiewb-netbeans-plugins-eclipse-formatter.enableFormatAsSaveAction>
-        <de-markiewb-netbeans-plugins-eclipse-formatter.useProjectSettings>true</de-markiewb-netbeans-plugins-eclipse-formatter.useProjectSettings>
-    </properties>
-</project-shared-configuration>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (junichi11) [inline nb-configuration.xml]: It seems that the line ending is changed.

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java [hunk 5]
```diff
@@ -163,9 +164,49 @@ private static String readFirstLineWithSeparator(FileObject fo, Charset charset)
...
+  public static String replaceLineEndings(String text, String lineEnding) {
+    List<String> lines = readLines(text);
+    return replaceLineEndings(lines, lineEnding);
+  }
+
+  public static List<String> readLines(String text) {
+    List<String> lines = new ArrayList<>();
+    try (BufferedReader reader = new BufferedReader(new StringReader(text))) {
+
+      try {
```
Comment (junichi11) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: I think that this `try` statement and `reader.close()` can be removed.

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java [hunk 6]
```diff
@@ -163,9 +164,49 @@ private static String readFirstLineWithSeparator(FileObject fo, Charset charset)
...
+        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
```
Comment (junichi11) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: Maybe, `while` statement is preferred. 
@bennyn What do you think?

**Comment by markiewb** [comment]
@junichi11 / @bennyn / @Shredder121 : Feel free to checkout the branch and fix the issues for yourself, if they are only minor ones.

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java (hunk 6 above)
Comment (Shredder121) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: I think nested expressions in `while` statements are not that readable.
Personally I am more of a for loop through `BufferedReader.lines` kinda guy.
But I won't stop you if you insist on it.

**Comment by junichi11** [comment]
@markiewb Thanks for your PR. Please don't misunderstand. As Benny wrote the above, it will be reviewed by @bennyn. I just left comments :) So, please wait a while.

**Comment by markiewb** [comment]
@junichi11: I do not mind and I do not misunderstand it. It's fine. I guess you are busy too. *g*

**Comment by bennyn** [comment]
@markiewb Sorry for being so late with reviewing it! But as you said, everyone's busy. I finally reviewed your changes, so you get a big DANKESCHÖN! :smile: 

I will add the `try-with-resources` blocks suggested by @junichi11 and @Shredder121 when this PR is merged because I don't have access to your fork. 

Next thing will then be plugin certification. Thank you so much @markiewb. :beers:

File: src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java (hunk 2 above)
Comment (bennyn) [inline src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java]: We can check how it's done in Apache Commons' `IOUtils.readLines`.

**Comment by markiewb** [comment]
@bennyn: Gern geschehen! Only with a public release and the placement at the plugin center your plugin gets the attention of the users. Your invested work should not be wasted, that's why I helped you.

**Pull request merged**